internal/kubernetes/event/event.go

```diff
@@ -140,3 +144,93 @@ func (e *sink) Patch(event *corev1.Event, data []byte) (*corev1.Event, error) {
 	err := e.client.Patch(e.ctx, event, libClient.RawPatch(types.StrategicMergePatchType, data))
 	return event, err
 }
+
+type AppReference struct {
```

Comment:
Is there any reason not to use a `types.NamespacedName` here?

Reply:
`types.NamespacedName` does not omit an empty namespace value. If the namespace is empty, it will be something like:
```
{
  "Namespace" : "",
  "Name" : "app-name"
}
```
instead of 
```
{
  "Name" : "app-name"
}
```

Reply:
Is it desirable for the namespace to be empty? I would think that in terms of traceability, it would be good to always include the namespace as it will otherwise be hard to derive the default without knowing how the controller is further configured.

Reply:
Makes sense to me. Revised to `types.NamespacedName` with a default namespace value to provide better understanding on the namespace without any additional context.